graphix/clifford.py

```diff
@@ -287,6 +293,17 @@ def measure(self, pauli: graphix.pauli.Pauli) -> graphix.pauli.Pauli:
         symbol, sign = table[pauli.symbol.value]
         return pauli.unit * graphix.pauli.TABLE[symbol + 1][sign][False]
 
+    @classmethod
+    def __get_pydantic_core_schema__(
+        cls, source_type: typing.Any, handler: pydantic.GetCoreSchemaHandler
+    ) -> pydantic_core.CoreSchema:
+        def check_clifford(obj) -> Clifford:
+            if not isinstance(obj, Clifford):
+                raise ValueError("Clifford expected")
+            return obj
+
+        return pydantic_core.core_schema.no_info_plain_validator_function(function=check_clifford)
```

Comment:
are we sure we want to use this, it does not seem to be the most stable feature:
https://docs.pydantic.dev/latest/concepts/types/#customizing-validation-with-__get_pydantic_core_schema__

Reply:
This change allows the `Clifford` class to be used as a type in `pydantic` models: in this PR, we changed the `cliff_index` parameter of type `int` in `C` commands into a `clifford` parameter of type `Clifford`. I think that having a parameter of type `Clifford` instead of a raw index is a good thing. However, another option is to enable the config flag `arbitrary_types_allowed` in the `C` command (https://docs.pydantic.dev/latest/api/config/#pydantic.config.ConfigDict.arbitrary_types_allowed): I don't know if it is more stable, but it is simpler indeed.

Reply:
> `Clifford` class to be used as a type in pydantic models

Why is it necessary? I feel it's very unlikely that `Clifford` should always be monitored by `pydantic` validation.

Reply:
Even if it were to be really necessary, we should not rely on internal APIs as @shinich1 insists.

Reply:
Fixed in cba7cf3: I finally set the flag `ConfigDict(arbitrary_types_allowed=True)` in the pydantic model for `C` commands, which makes pydantic do the validation with `isinstance` anyway. There is a similar discussion [here](https://github.com/TeamGraphix/graphix/pull/158#discussion_r1722512340).